File: src/main/java/com/meti/Input.java
package com.meti;

import java.util.Objects;

public class Input {
	private final String content;

	public Input(String content) {
		this.content = content;
	}

	public int firstChar(char c) {
		return getContent().indexOf(c);
	}

	public String getContent() {
		return content;
	}

	@Override
	public int hashCode() {
		return Objects.hash(content);
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;
		Input input = (Input) o;
		return Objects.equals(content, input.content);
	}

	@Override
	public String toString() {
		return "{\"content\":\"%s\"}".formatted(content);
	}

	public Input slice(int from) {
		return slice(from, content.length());
	}

	Input slice(int from, int to) {
		var typeString = getContent().substring(from, to).trim();
		return new Input(typeString);
	}
}

File: src/main/java/com/meti/Input.scala
package com.meti

class Input(val input: String) {
  def asOutput = new StringOutput(getInput)

  def isEmpty = getInput.isBlank

  def getInput = input

  def slice(extent: Int) = {
    val slice = getInput.substring(extent)
    val value = slice.trim
    new Input(value)
  }

  def stream: Stream[Char] = Strings.stream(getInput)

  def test = getInput.startsWith("return ")
}